Homework2/DiagonalMatrix/MatrixExtensions.cs

```diff
@@ -0,0 +1,47 @@
+﻿
+namespace DiagonalMatrix
+{
+    public static class MatrixExtensions
+    {
+        public static Matrix AddMatrix(this Matrix matrix, Matrix matrixToAdd)
+        {
+            var summedElements = new int[0];
```

Comment:
Lets simplify the code below. 

   int newMatrixSize = matrix.Size >= matrixToAdd.Size ? matrix.Size : matrixToAdd.Size;
    int[] newMatrixDiagonalElements = new int[newMatrixSize];

    for ( int i = 0; i < newMatrixSize; i++ )
    {
        newMatrixDiagonalElements[i] = matrix[i, i] + matrixToAdd[i, i]; 
    }

    return new DiagonalMatrix(newMatrixDiagonalElements);